src/terminal7.js

```diff
@@ -839,7 +816,7 @@ peer_name = "${peername}"\n`
             const gate = gatePad.gate
             if (!this.longPressGate && gate)
                 this.longPressGate = this.run(() => {
-                    gate.edit()
+                    this.map.shell.runCommand("edit", [gate.name])
```

Comment:
we already have the gate and we pass the name to later be used to find the gate. A better function will be one that can accept either a `string` with the name of the gate or the `Gate` itself

Reply:
Putting typescript syntax a side for a minute, all commands that accepts a gate as their arg can start with:
``` 
const gate = (arg[0] instanceof Gate)?arg[0]:shell.getGate(arg[0])
```
In typescript we might have to define an `Arg` type as `string | Gate ` and we can add any other objects that could reach the command.